MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/labs/RecyclerViewOnMapActivity.java

```diff
@@ -0,0 +1,275 @@
+import com.mapbox.mapboxsdk.geometry.LatLng;
+import com.mapbox.mapboxsdk.maps.MapView;
+import com.mapbox.mapboxsdk.maps.MapboxMap;
+import com.mapbox.mapboxsdk.maps.OnMapReadyCallback;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Use a recyclerview with a Mapbox map to easily explore content all on one screen
+ */
+public class RecyclerViewOnMapActivity extends AppCompatActivity {
+
+  private MapView mapView;
+  public MapboxMap mapboxMap;
+  private RecyclerView recyclerView;
+  private LocationRecyclerViewAdapter locationAdapter;
+  private ArrayList<SingleRecyclerViewLocation> locationList;
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+
+    // Mapbox access token is configured here. This needs to be called either in your application
+    // object or in the same activity which contains the mapview.
+    Mapbox.getInstance(this, getString(R.string.access_token));
+
+    // This contains the MapView in XML and needs to be called after the access token is configured.
+    setContentView(R.layout.activity_lab_recycler_view_on_map);
+
+    recyclerView = (RecyclerView) findViewById(R.id.rv_on_top_of_map);
+
+    Toast.makeText(this, R.string.toast_instruction, Toast.LENGTH_SHORT).show();
+
+    // Initialize the map view
+    mapView = (MapView) findViewById(R.id.mapView);
+    mapView.onCreate(savedInstanceState);
+    mapView.getMapAsync(new OnMapReadyCallback() {
+      @Override
+      public void onMapReady(MapboxMap mapboxMap) {
+        RecyclerViewOnMapActivity.this.mapboxMap = mapboxMap;
+
+        setUpLists();
+
+        // Set up the recyclerView
+        locationAdapter = new LocationRecyclerViewAdapter(locationList, mapboxMap);
+        recyclerView.setLayoutManager(new LinearLayoutManager(getApplicationContext(),
+            LinearLayoutManager.HORIZONTAL, true));
+        recyclerView.setItemAnimator(new DefaultItemAnimator());
+        recyclerView.setAdapter(locationAdapter);
+      }
+    });
+
+  }
+
+  // Add the mapView lifecycle to the activity's lifecycle methods
+  @Override
+  public void onResume() {
+    super.onResume();
+    mapView.onResume();
+  }
+
+  @Override
+  protected void onStart() {
+    super.onStart();
+    mapView.onStart();
+  }
+
+  @Override
+  protected void onStop() {
+    super.onStop();
+    mapView.onStop();
+  }
+
+  @Override
+  public void onPause() {
+    super.onPause();
+    mapView.onPause();
+  }
+
+  @Override
+  public void onLowMemory() {
+    super.onLowMemory();
+    mapView.onLowMemory();
+  }
+
+  @Override
+  protected void onDestroy() {
+    super.onDestroy();
+    mapView.onDestroy();
+  }
+
+  @Override
+  protected void onSaveInstanceState(Bundle outState) {
+    super.onSaveInstanceState(outState);
+    mapView.onSaveInstanceState(outState);
+  }
+
+  public boolean onOptionsItemSelected(MenuItem item) {
+    switch (item.getItemId()) {
+      case android.R.id.home:
```

Comment:
in a different PR I added back `android:launchMode="singleTop"` which is what we used in the past for this issue. the `onOptionsItemSelected` boilerplate can now be removed.